crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -125,6 +125,94 @@ impl Ellipse {
     }
 }
 
+/// A primitive shape formed by the region between two circles, also known as a ring.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[doc(alias = "Ring")]
+pub struct Annulus {
+    /// The inner radius of the annulus
+    pub inner_radius: f32,
+    /// The outer radius of the annulus
+    pub outer_radius: f32,
+}
+impl Primitive2d for Annulus {}
+
+impl Default for Annulus {
+    /// Returns the default [`Annulus`] with radii of `0.5` and `1.0`.
+    fn default() -> Self {
+        Self {
+            inner_radius: 0.5,
+            outer_radius: 1.0,
+        }
+    }
+}
+
+impl Annulus {
+    /// Create a new [`Annulus`] from an inner and outer radius
+    #[inline(always)]
+    pub const fn new(inner_radius: f32, outer_radius: f32) -> Self {
+        Self {
+            inner_radius,
+            outer_radius,
+        }
+    }
+
+    /// Get the diameter of the annulus
+    #[inline(always)]
+    pub fn diameter(&self) -> f32 {
+        2.0 * self.outer_radius
+    }
+
+    /// Get the width of the annulus
+    #[inline(always)]
+    pub fn width(&self) -> f32 {
+        self.outer_radius - self.inner_radius
+    }
+
+    /// Get the area of the annulus
+    #[inline(always)]
+    pub fn area(&self) -> f32 {
+        PI * (self.outer_radius.powi(2) - self.inner_radius.powi(2))
+    }
+
+    /// Get the perimeter or circumference of the annulus,
+    /// which is the sum of the outer and inner edges.
```

Comment:
Calling them "edges" isn't really accurate since geometrically [edges are line segments](https://en.wikipedia.org/wiki/Edge_(geometry)).

I think "perimeter" is a common enough term that we can just change it to this?

```suggestion
    /// Get the perimeter or circumference of the annulus,
    /// which is the sum of the perimeters of the inner and outer circles.
```